### PR TITLE
Documentation Overhaul

### PR DESCRIPTION
**Pull request by mDuo13**
This expands and improves the APIDoc.js documentation from this project:

- It adds a new intro Markdown file, with object parameter tables for all data types returned by the API.
    - Parameter tables are dynamically generated from JSON schemas by a fork of ripple-lib's documentation toolchain.
- I rewrote all the API method explanations to be more concise, accurate, and consistent. Same for the example requests and responses.
- I added summaries of the response format where there weren't any.
- Updated README to describe the `npm run docs` script. Moved the environment variable listing to the generated documentation.

### CODE REVIEW

File: src/controllers/transfers.js [hunk 1]
```diff
@@ -9,7 +9,43 @@ const model = require('../models/transfers')
 const InvalidUriParameterError = require('five-bells-shared/errors/invalid-uri-parameter-error')
 
 /**
- * @api {get} /transfers/:id Get local transfer object
+ * @apiGroup Test Group
+ * @apiDescription lorem ipsum dolor sit amet
```
Comment (emschwartz) [inline src/controllers/transfers.js]: ?

File: README.md [hunk 2]
```diff
@@ -33,7 +34,7 @@ npm install
 To run it using an in-memory database (the simplest option), run:
 
 ``` sh
-LEDGER_DB_SYNC=1 LEDGER_DB_URI=sqlite://:memory: npm start
+LEDGER_ADMIN_PASS=mypassword LEDGER_DB_SYNC=1 LEDGER_DB_URI=sqlite://:memory: npm start
```
Comment (emschwartz) [inline README.md]: Is this option necessary? Why not use the default?

**Comment by emschwartz** [comment]
LGTM. I'm guessing the build is failing the integration tests because of the latest `ilp-core` changes

File: src/controllers/transfers.js (hunk 1 above)
Comment (mDuo13) [inline src/controllers/transfers.js]: That was a test to see if I could create groups without associated API methods, which snuck into an in-progress commit. It got removed when it became clear that you cannot define groups that aren't associated with API methods.

File: README.md (hunk 2 above)
Comment (mDuo13) [inline README.md]: By default, there is no admin password, which, in my experience, makes it impossible to authenticate as an admin if you run the ledger using the command here as written previously.

I'm glad to be overruled if that's my misunderstanding.

**Comment by mDuo13** [comment]
One of the test failures is eslint for the new doc build-scripts. I'll see if I can fix that.

**Comment by mDuo13** [comment]
OK. I've fixed the test failures that seem to be based on my changes. I'm going to assume the other failures are due to unrelated ilp-core changes.

**Comment by bensharaf** [comment]
https://circleci.com/gh/interledger/five-bells-ledger/1380 seems to be failing because of the syntax used in `five-bells-ledger/scripts/json-schema-to-markdown-table2.js:210`

**Pull request merged**